### PR TITLE
Add deployment demo

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: tools/docker/deploy/Dockerfile_template.centos8.1 [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# base stage:
+# Install dependencies for deployment to get minimum size for deployer.
+# For deployment environment, only occlum-runtime, sgx-psw and python2 are needed.
```
Comment (guzongmin) [inline tools/docker/deploy/Dockerfile_template.centos8.1:3]: We may need to update python2 to python3.

File: tools/docker/deploy/Dockerfile_template.centos8.1 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+>> /etc/yum.repos.d/inclavare-containers.repo && \
+    yum install --nogpgcheck -y python2 occlum-runtime && \
+    yum clean all && \
+    ln -s /usr/bin/python2 /usr/local/bin/python && \
+    mkdir -p /var/run/aesmd
+ENV PATH="/opt/occlum/build/bin:/usr/local/occlum/bin:$PATH"
+
+
+# packager stage:
+# Users can build their own applications and put to occlum instance. And then use "occlum package"
```
Comment (guzongmin) [inline tools/docker/deploy/Dockerfile_template.centos8.1:24]: Is it better "Build your own application and ..."?

File: tools/docker/deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+With the help of [docker multistage build](https://docs.docker.com/develop/develop-images/multistage-build/) and `occlum package` command, we provide dockerfile templates to build a image with the smallest size for deployment environment.
+
+Checkout the dockerfile templates for [Ubuntu](./Dockerfile_template.ubuntu18.04) and [CentOS](./Dockerfile_template.centos8.1). There are three stages in each dockerfile:
+
+    - base stage: This stage configures the package management systems of specific OS and intall required packages for deployment, including `occlum-runtime`, sgx-psw packages and python. If users want to install specific version of packages, modification should be done in this stage.
+
+    - packager stage: This stage is to build and package the application for deployment. Users can build their own applications and put to occlum instance. And then use "occlum build" and "occlum package" commands to get a minimum subset of files to run in deployment environment. To support full Occlum commands, extra dependencies are installed.
```
Comment (guzongmin) [inline tools/docker/deploy/README.md:11]: Should highlight: User should finish the enclave signing in this stage.

**Comment by tatetian** [comment]
I think this commit can serve our users better by rewriting it as a sample project for deployment, demonstrating how to shift from development phase to deployment phase using a simple app (e.g., helloworld). This demo can also be added as a CI test (that will be triggered if new versions of installation packages are generated) so that the `occlum package` command the Occlum installation packages can be tested automatically through CI.

**Comment by jessehui** [comment]
Add deployment template as demo and add to hardware CI test.
Results can be checked here: https://github.com/jessehui/occlum/actions/runs/1047575724

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**